main.js

```diff
@@ -1,115 +1,115 @@
-// let names = [];
-// let numbers = [];
+let contacts = [];
 
-function getValue (fieldName) {
+function getValue(fieldName) {
     return document.getElementById(fieldName).value;
 }
 
-function run() {
-    const submitButton = document.getElementById('submitButton');
-    // const bodyTable = document.getElementById('bodyTable');
-
-    // function validate(name, number) {
-
-    // }
-
-    function renderTable(contacts) {
-        const contactList = document.getElementById('bodyTable');
-        contactList.innerHTML = "";
-        contacts.forEach(
-            ({ name, number }) => {
-                contactList.innerHTML += 
-                    `<tr>
-                        <td>${name}</td>
-                        <td>${number}</td>
-                    </tr>`;
-            }
-        )
-    }
-
-    function readFormAndAddNewLine(){
-        // const name = getValue('name');
-        // const number = getValue('number');
-        // const validationResults = validate(name, number);
-        // error = { fieldName: null, message: 'Esse contato já existe'}
-        // error = { fieldName: 'name', message: 'Já existe um contato com esse nome ....'}
-        // // fieldName:name|null message
-        // console.log({ name, number });
-        const contacts = [
-            {name: "João", number: "1111111"},
-            {name: "Bruno", number: "22222222"},
-            {name: "Paulo", number: "3333333"},
-        ];
-        renderTable(contacts)
-
-    
-        // document.getElementById('name').value = "";
-        // document.getElementById('number').value = "";
-    }
-     
-    //console.log(submitButton);
-    submitButton.addEventListener('click', readFormAndAddNewLine)
+function highlight() {
+    const existingNames = document.querySelectorAll('.name');
+    const existingNumbers = document.querySelectorAll('.number');
+    existingNames.forEach(existingName => {
+        if (existingName.innerHTML === getValue('name')) {
+            existingName.classList.add('highlight');
+        }
+    })
+    existingNumbers.forEach(existingNumber => {
+        if (existingNumber.innerHTML === getValue('number')) {
+            existingNumber.classList.add('highlight');
+        }
+    })
 }
 
-run();
+function undoHighlight() {
+    const highlighted = document.querySelectorAll('.highlight');
+    highlighted.forEach(element => element.classList.remove('highlight'));
+};
 
+function validate(contacts, formName, formNumber) {
 
+    const errors = { name: 0, number: 0 };
+    const sameName = ({ name }) => name == formName
+    const sameNumber = ({ number }) => number == formNumber
 
+    if (contacts.find(sameName)) {
```

Comment:
Isso aqui está repetido (linhas 33 a 36 e 38 a 41). Só a função e a mensagem de erro que são diferentes. É importante ir treinando o olho para reconhecer padrões.

Reply:
de maneira geral, pensando em separar dados da renderização eu faria o seguinte: 
1 - pegar todas as mensagens de erro
2 - mostrar todos os erros.

Reply:
O problema quando você faz isso, além da repetição do código, é que deixa de ter uma "single source of truth". 
Se um código ficar diferente do outro, o programa vai ficar inconsistente. Há o risco de mostrar os hightlights  sem os textos dos erros e vice-versa.